app/components/views/Offset/index.tsx

```diff
@@ -610,19 +620,41 @@ export const Offset = (props: Props) => {
                 min={paymentMethod === "fiat" ? "1" : "0"}
                 value={quantity}
                 onKeyDown={(e) => {
+                  // don't allow decimal values when fiat payment method is selected
+                  if (paymentMethod === "fiat" && ["."].includes(e.key)) {
+                    e.preventDefault();
+                    setIsInvalidFractionalQtyFiat(true);
+                  }
                   // dont let user enter these special characters into the number input
                   if (["e", "E", "+", "-"].includes(e.key)) {
                     e.preventDefault();
                   }
                 }}
-                data-error={invalidCost || invalidRetirementQuantity}
+                data-error={
+                  invalidCost ||
+                  invalidRetirementQuantity ||
+                  isInvalidFractionalQtyFiat
+                }
                 onChange={handleChangeQuantity}
+                onBlur={() => setIsInvalidFractionalQtyFiat(false)}
                 placeholder={t({
                   id: "offset.offset_quantity",
                   message: "Enter quantity to offset",
                 })}
               />
             </div>
+
+            {isInvalidFractionalQtyFiat && paymentMethod === "fiat" && (
+              <Text
+                t="caption"
+                color="lightest"
+                className="invalid_project_tonnage"
+              >
+                <Trans id="offset.invalid_fractional_quantity_fiat">
+                  Cannot use a fractional quantity when paying with credit card!
```

Comment:
I wonder if we can come up with a more user friendly error messaging here.
"Fractional" isn't immediately clear to me.

@ShimonD-KlimaDAO You're good with this kind of stuff. Could you please give us a hand?

Using the error message format for the [pledge form](https://github.com/KlimaDAO/klimadao/blob/staging/site/components/pages/Pledge/lib/formSchema.ts) as a reference.
```
Enter a whole number when paying with a credit card
```

Reply:
@0xAeterno Firstly, thank you for the feedback and warm welcome. It's great to be here and I'm excited to be able to contribute  :)

Agreed on this, I wasn't sure what the error message should be here - but your suggestion is much better than what I have currently there. Will make the change once we get confirmation from @ShimonD-KlimaDAO. :)